docs/sphinx/source/jdbc/advanced.rst

```diff
@@ -0,0 +1,224 @@
+======================
+Advanced JDBC Features
+======================
+
+.. important::
+   The JDBC interface is experimental and not production-ready at this stage. APIs and behaviors are subject to change.
+
+This guide covers FoundationDB Record Layer-specific JDBC features for working with complex data types like STRUCTs and ARRAYs.
+
+.. note::
+   **Driver-Specific Classes**: When creating STRUCT and ARRAY values, use the appropriate builder for your JDBC driver:
+
+   - **Embedded Driver**: ``EmbeddedRelationalStruct`` and ``EmbeddedRelationalArray`` (from ``com.apple.foundationdb.relational.api``)
+   - **Server Driver**: ``JDBCRelationalStruct`` and ``JDBCRelationalArray`` (from ``com.apple.foundationdb.relational.jdbc``)
```

Comment:
Is this true? Do we not have an API that works for both the embedded and server use cases? I would have expected that there'd be something like a `RelationalStruct` object that the user could create (in the API package) and then something could translate that to the appropriate one

Reply:
We do have a common interface which is `RelationalStruct` and `RelationalArray`, and respective builders for the EmbeddedDriver and the ServerDriver. I'll adjust the documentation. Beyond the creation of the builder, everything else is the same.

Reply:
I agree that we probably would want to build something more generic in the future so that we push down the conversion at the lower level.